src/manifolds/GeneralizedStiefel.jl

```diff
@@ -0,0 +1,242 @@
+@doc doc"""
+    GeneralizedStiefel{n,k,T} <: AbstractEmbeddedManifold{AbstractIsometricEmbeddingType}
```

Comment:
```suggestion
    GeneralizedStiefel{n,k,T} <: AbstractEmbeddedManifold{AbstractEmbeddingType}
```
We could show that it's an isometric embedding in a metric manifold (Euclidean with metric given by the metric B)?

By the way, I feel like we should always give a concrete type here (that's what I meant earlier by `DefaultEmbeddingType` and `DefaultIsometricEmbeddingType` and why I was confused as to what it meant before the name was changed).